python_modules/libraries/dagster-aws/dagster_aws/s3/poll_upload.py

```diff
@@ -0,0 +1,44 @@
+import json
+import os
+import sys
+import time
+
+from dagster._serdes.serdes import deserialize_json_to_dagster_namedtuple
+from dagster._utils.interrupts import capture_interrupts, pop_captured_interrupt
+
+
+def current_process_is_orphaned(parent_pid):
+    parent_pid = int(parent_pid)
+    if sys.platform == "win32":
+        import psutil  # pylint: disable=import-error
+
+        try:
+            parent = psutil.Process(parent_pid)
+            return parent.status() != psutil.STATUS_RUNNING
+        except psutil.NoSuchProcess:
+            return True
+
+    else:
+        return os.getppid() != parent_pid
```

Comment:
do we not have a similar thing defined in a more core place? I know we got an external PR to fix-up similar code in the past